sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2919,6 +2919,17 @@ object SQLConf {
       .stringConf
       .createWithDefault("csv,json,orc,parquet")
 
+  val ADD_DEFAULT_COLUMN_EXISTING_TABLE_BANNED_PROVIDERS =
```

Comment:
It seems that having DEFAULT_COLUMN_ALLOWED_PROVIDERS is enough. Why do we need to have a new one here?

Reply:
Good Q: we want to differentiate between which table providers can allow column DEFAULT values with all commands, vs. which ones support all commands *except* `ALTER TABLE ADD COLUMN`. The reason is that the latter command requires the corresponding data source to implement specific support for filling in the existence default values during future scans, while all other commands simply use the `ResolveDefaultColumns` analysis rule in a generic way.

We've implemented this data source scan support for CSV, JSON, Orc, and Parquet file formats, but not for any V2 data sources yet. If someone wants to extend a V2 data source to support column DEFAULT values, they'll have to block `ALTER TABLE ADD COLUMN` commands at first. This PR provides a way to do that.